core/storage/suggestion/gae_models.py

```diff
@@ -72,6 +72,11 @@
 # The delimiter to be used in score category field.
 SCORE_CATEGORY_DELIMITER = '.'
 
+ALLOWED_QUERY_FIELDS = ['suggestion_type', 'target_type', 'target_id',
```

Comment:
I might be missing something here. Are you planning to have users query by author ID, assigned reviewer ID, final reviewer ID directly? I don't see how users will know the user ID...

Also, what is the context for these searches? Where will they happen from?

Reply:
Possible query situations I can think of are -- all suggestions by me, assigned to me, assigned to user XYZ, reviewed by user XYZ, my rejected suggestions, suggestions in category X, translation suggestions in language X, etc. I'm not entirely sure how much of this we will be implementing e2e. But just making sure it is extensible for all of these situations. 

As for the IDs part. The most useful situation I think is query stuff linked to me. so in these situations, we will pass in the user ID of the logged in user from the frontend. As for searching by user ID I agree it is not possible for the user to know the ID. We could (in the frontend) display user ids or search by user ids and pass the corresponding ID through the URL.

There isn't a fixed context for this function. The motivation was to not have to write a separate function everytime we discover a new context to list out suggestions. Like say, we were talking about having to display only my rejected suggestions in the creator dashboard. So we could use this there. Also in the future, if we do develop something like the github PRs page for contributors to view suggestions, then we could have dropdowns/fliters for allowing the reviewers to find specific suggestions easily.